fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java

```diff
@@ -82,48 +112,62 @@ private LogicalPlan bindWithCurrentDb(CascadesContext cascadesContext, UnboundRe
         }
-        Table table = cascadesContext.getTable(dbName, nameParts.get(1), connectContext.getEnv());
-        List<Long> partIds = getPartitionIds(table, unboundRelation);
-        if (table.getType() == TableType.OLAP) {
-            if (!CollectionUtils.isEmpty(partIds)) {
-                return new LogicalOlapScan(RelationUtil.newRelationId(), (OlapTable) table,
-                        ImmutableList.of(dbName), partIds);
-            } else {
-                return new LogicalOlapScan(RelationUtil.newRelationId(), (OlapTable) table, ImmutableList.of(dbName));
-            }
-        } else if (table.getType() == TableType.VIEW) {
-            Plan viewPlan = parseAndAnalyzeView(table.getDdlSql(), cascadesContext);
-            return new LogicalSubQueryAlias<>(table.getName(), viewPlan);
+        String tableName = nameParts.get(1);
+        TableIf table = getTable(catalogName, dbName, tableName, connectContext.getEnv());
+        return getLogicalPlan(table, unboundRelation, dbName, cascadesContext);
+    }
+
+    private LogicalPlan bindWithCatalogNameFromNamePart(CascadesContext cascadesContext,
+                                                        UnboundRelation unboundRelation) {
+        List<String> nameParts = unboundRelation.getNameParts();
+        ConnectContext connectContext = cascadesContext.getConnectContext();
+        String catalogName = nameParts.get(0);
+        String dbName = nameParts.get(1);
+        if (!dbName.equals(connectContext.getDatabase())) {
+            dbName = connectContext.getClusterName() + ":" + dbName;
+        }
+        String tableName = nameParts.get(2);
+        TableIf table = getTable(catalogName, dbName, tableName, connectContext.getEnv());
+        return getLogicalPlan(table, unboundRelation, dbName, cascadesContext);
+    }
+
+    private LogicalPlan getLogicalPlan(TableIf table, UnboundRelation unboundRelation, String dbName,
+                                       CascadesContext cascadesContext) {
+        switch (table.getType()) {
+            case OLAP:
+                List<Long> partIds = getPartitionIds(table, unboundRelation);
+                if (!CollectionUtils.isEmpty(partIds)) {
+                    return new LogicalOlapScan(RelationUtil.newRelationId(),
+                        (OlapTable) table, ImmutableList.of(dbName), partIds);
+                } else {
+                    return new LogicalOlapScan(RelationUtil.newRelationId(),
+                        (OlapTable) table, ImmutableList.of(dbName));
+                }
+            case VIEW:
+                Plan viewPlan = parseAndAnalyzeView(((View) table).getDdlSql(), cascadesContext);
+                return new LogicalSubQueryAlias<>(table.getName(), viewPlan);
+            case HMS_EXTERNAL_TABLE:
+                return new LogicalFileScan(cascadesContext.getStatementContext().getNextRelationId(),
+                    (HMSExternalTable) table, ImmutableList.of(dbName));
```

Comment:
I suggest rename LogicalFileScan to LogicalExternalRelation(with field: HMSExternalTable table) or LogicalHMSTable

Reply:
I'll take a look at this, probably fix it in next PR.